### ISSUE TITLE
Improved liveness probe for nfd-master

### ISSUE THREAD

**Issue by marquiz** [kubernetes-sigs/node-feature-discovery#1548]
**What would you like to be added**:

The implementation of the gRPC health server in nfd-master is very simple and naive, in practice only telling that the nfd-master main loop was started and that the program hasn't exited. This could be improved e.g. by including the state of the metrics server, nfd api controller and node updater pool (i.e. healthcheck fails if any of those goes down).

**Why is this needed**:

More trust on the liveness data.

**Comment by omerap12** [comment]
Hey @marquiz  , maybe I can assist with it :)

**Comment by marquiz** [comment]
> Hey @marquiz , maybe I can assist with it :)

Wow, appreciated

**Comment by omerap12** [comment]
> > Hey @marquiz , maybe I can assist with it :)
> 
> Wow, appreciated

sure why not, seems like a task I can take ( but I may ask a few questions along the way if that's ok )

**Comment by marquiz** [comment]
> I may ask a few questions along the way if that's ok

Definitely. The "design" details in the description are pretty vague and really hastily written

**Comment by omerap12** [comment]
Hi @marquiz, I've started working on issue #1814. Could you take a quick look and let me know if I'm on the right track?

**Comment by omerap12** [comment]
friendly remainder :) @marquiz

**Comment by omerap12** [comment]
> > friendly remainder :) @marquiz
> 
> Thanks @omerap12. I try to find a peaceful slot tomorrow to look at that

Sure, it's just a quick sketch, and I wanted to check if I'm addressing this in a sensible manner. Thanks!
